src/rustdoc_gen.rs

```diff
@@ -110,6 +110,65 @@ impl<'a> CrateSource<'a> {
         all_crate_features.extend(self.regular_features());
         all_crate_features.into_iter().collect()
     }
+
+    /// Some feature names usually mean, that the feature is breaking semver by design.
+    /// This function filters them out.
+    ///
+    /// Feature names not included by this function:
+    ///  - `unstable`
+    ///  - `nightly`
+    ///  - `bench`
+    ///  - `no_std`
+    ///  - features with prefix `__`
+    fn heuristically_included_features(&self) -> Vec<String> {
+        let features_ignored_by_default = std::collections::HashSet::from([
+            String::from("unstable"),
+            String::from("nightly"),
+            String::from("bench"),
+            String::from("no_std"),
+        ]);
+
+        let determine = |feature_name: &String| {
+            !features_ignored_by_default.contains(feature_name) && !feature_name.starts_with("__")
+        };
+
+        self.all_features().into_iter().filter(determine).collect()
+    }
+
+    pub(crate) fn feature_list_from_config(
```

Comment:
Can you document what this function does? It has some fairly complex logic inside and I don't really understand what's going on or what it returns in each of the edge cases inside.